NotifyMe/services/service.py

```diff
@@ -0,0 +1,202 @@
+
+import logging
+from django.utils import timezone
+from datetime import timedelta
+
+from django.db import IntegrityError
+from rest_framework.exceptions import ValidationError
+from NotifyMe.constants import plans, plans_duration
+from django.core.exceptions import ObjectDoesNotExist, MultipleObjectsReturned
+from NotifyMe.models.notification import Notification
+from NotifyMe.models.notificationType import NotificationType
+from NotifyMe.models.subscription import Subscription
+from NotifyMe.models.subscriptionPlan import SubscriptionPlan
+from NotifyMe.models.user import User
+from django.db import DatabaseError
+
+logger = logging.getLogger(__name__)
+
+class UserService:
+    def get_user_data(self, request):
+        try:
+            users = User.objects.all()
+            return users
+        except ObjectDoesNotExist as e:
+            logger.error(f"Object does not exist error: {e}")
+            return None
+        except MultipleObjectsReturned as e:
+            logger.error(f"Multiple objects returned error: {e}")
+            return None
+        except DatabaseError as e:
+            logger.error(f"Database error: {e}")
+            return None
+        except Exception as e:
+            logger.error(f"General error: {e}")
+            return None
+        
+        
+    def get_userId_data(self, data):
+        user = User.objects.get(id=data.get('id'))
+        return user
+    
+    
+    def get_userDatabase(self):
+        return User
+    
+    def get_subscriptionPlanDatabase(self):
```

Comment:
what is the use of this method

Reply:
for handling model.DoesNotExists exceptions